bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/internal/SWTFontProvider.java

```diff
@@ -26,20 +27,45 @@
  * take the provided values for the zoom into consideration and return scaled variant of a font if necessary.
  */
 public class SWTFontProvider {
-	private static Map<Device, SWTFontRegistry> fontRegistries = new HashMap<>();
+	private static Map<Device, SWTFontRegistry> fontRegistries = new ConcurrentHashMap<>();
```

Comment:
According to the [javadoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/concurrent/ConcurrentHashMap.html)

> Like [Hashtable](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Hashtable.html) but unlike [HashMap](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/HashMap.html), this class does not allow null to be used as a key or value.

So wen should add null checks at the getters for `display` below

Reply:
It seems slightly odd that it's not final.    I know it wasn't final before and that it doesn't really matter...

Reply:
Indeed, just changed it to final.